javascript/javascript1/week1/aDogAgeCalculator.js

```diff
@@ -0,0 +1,12 @@
+// Goodboy-Oldboy (A dog age calculator)
+let dogYearFuture = 2027;
+let dogYearOfBirth = 2017;
+let dogHumanYear = dogYearFuture - dogYearOfBirth;
+let dogYear = dogHumanYear * 7;
+let shouldShowResultInDogYears = true;
+if (shouldShowResultInDogYears = true) {
```

Comment:
This is probably the most common and most annoying bug ever in coding. I have made this mistake many times when I was a beginner. If you try to change the value of `shouldShowResultInDogYears` to `false`, it still prints it in dog years. Why???

I can tell you why. 😉 The condition to check if two values are equal is `a == b`, or `a === b` if you care about the type as well. However, you wrote `shouldShowResultInDogYears = true` instead of `shouldShowResultInDogYears == true` or `shouldShowResultInDogYears === true`. 

But why does JavaScript not give me an error? If I have written something incorrect?! Well, the worst thing is: you did not. You wrote something perfectly correct: you assigned the value `true` to the variable `shouldShowResultInDogYears`, and then you just check `if shouldShowResultInDogYears`. However, `shouldShowResultInDogYears` is now equal to `true`, so this `if` statement will always be `true`, and the `else` branch will never be executed! So annoying!!! 

So yes, make sure that you always have `==` or `===` in any condition. ☝️ And, one more tip, if you have a variable that is a `boolean`, so either `true` or `false`, it is also enough to just write:

```javascript
if shouldShowResultInDogYears {
  // some code
} else {
  // some other code
}
```